Time of the day - correction (issue/8247)

kingfisher88 opened the pull request:
Remove "anonymous (not verified)" from greeting message for anonymous users.
Change the service that returns the current time of day, returns just the
greeting message according to the time of day.

kingfisher88 committed:
Remove "anonymous (not verified)" from greeting message for anonymous users.
Change the service that returns the current time of day, returns just the
greeting message according to the time of day.

lekob commented on web/modules/custom/usergreeting/src/GreetingTime.php:
```diff
@@ -31,17 +31,10 @@ final class GreetingTime {
-   * @var \Drupal\usergreeting\Drupal\Core\StringTranslation\TranslationInterface
+   * @var \Drupal\Core\StringTranslation\TranslationInterface
    */
   protected $stringTranslation;
```
The $stringTranslation property is defined in StringTranslationTrait. No need to redefine it here.

gkurucz replied:
According to https://api.drupal.org/api/drupal/core!lib!Drupal!Core!StringTranslation!StringTranslationTrait.php/trait/StringTranslationTrait/8.8.x he should.

> If the class is capable of injecting services from the container, it should inject the 'string_translation' service and assign it to $this->stringTranslation.

I've also seen this exact same practice in core.

lekob commented on web/modules/custom/usergreeting/src/GreetingTime.php:
```diff
@@ -70,7 +60,7 @@ public function __construct(ConfigFactoryInterface $config_factory, TimeInterfac
    * @return string
```
You are not returning a string.

lekob commented on web/modules/custom/usergreeting/src/GreetingTime.php:
```diff
@@ -70,7 +60,7 @@ public function __construct(ConfigFactoryInterface $config_factory, TimeInterfac
+  public function greetingMessage(): TranslatableMarkup {
     $time_output = (int) date('G', $this->timeOutput->getRequestTime());
     $config_morning = $this->configFactory->get('usergreeting.settings')->get('morning_start');
     $config_afternoon = $this->configFactory->get('usergreeting.settings')->get('afernoon_start');
```
afternoon

lekob commented on web/modules/custom/usergreeting/src/Plugin/Block/UsergreetingBlock.php:
```diff
@@ -38,33 +46,52 @@ class UsergreetingBlock extends BlockBase implements ContainerFactoryPluginInter
   /**
    * {@inheritdoc}
    */
   public function build() {
```
Declare array return type.

kingfisher88 committed:
Remove defining  $stringTranslation protected varible in GreetingTime.php
Correct the returning type of greetingMessage()
Correct typos afernoon -> afternoon
Declare return type of build() in UsergreetingBlock.php

kingfisher88 committed:
Specifi the return object (TranslatableMarkup) for the greetingMessage()

kingfisher88 committed:
Fix code with PHP Code Beautifier.

gkurucz commented on web/modules/custom/usergreeting/src/Plugin/Block/UsergreetingBlock.php:
```diff
@@ -38,33 +46,52 @@ class UsergreetingBlock extends BlockBase implements ContainerFactoryPluginInter
-    $build['#cache']['max-age'] = 0;
-    return $build;
-
+    if ($this->account->isAnonymous() === TRUE) {
```
This condition can be simplified.

```
$message = [
  common_stuff...
];

if (condition) {
  $message['different_stuff'] = 'different_stuff';
  $message['stuff_to_change'] = 'other_different_stuff';
}
```

kingfisher88 committed:
Simplify the if condition in UsegreetingBlock.php
